pwndbg/commands/mprotect.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+
+import gdb
+import pwndbg.chain
+import pwndbg.commands
+import pwndbg.enhance
+import pwndbg.file
+import pwndbg.which
+import pwndbg.wrappers.checksec
+import pwndbg.wrappers.readelf
+from pwndbg.color import message
+
+parser = argparse.ArgumentParser(description='Calls mprotect. x86_64 only.')
+parser.add_argument('addr', help='Page-aligned address to all mprotect on.',
+                    type=int)
+parser.add_argument('length', help='Count of bytes to call mprotect on. Needs '
+                    'to be multiple of page size.',
+                    type=int)
+parser.add_argument('prot', help='Prot string as in mprotect(2). Eg. '
+                    '"PROT_READ|PROT_EXEC"', type=str)
+
+SYS_MPROTECT = 0x7d
+
+prot_dict = {
+    'PROT_NONE': 0x0,
+    'PROT_READ': 0x1,
+    'PROT_WRITE': 0x2,
+    'PROT_EXEC': 0x4,
+}
+
+def prot_str_to_val(protstr):
+    '''Heuristic to convert PROT_EXEC|PROT_WRITE to integer value.'''
+    prot_int = 0
+    for k in prot_dict:
+        if k in protstr:
+            prot_int |= prot_dict[k]
+    return prot_int
+
+
+@pwndbg.commands.ArgparsedCommand(parser)
+@pwndbg.commands.OnlyWhenRunning
+def mprotect(addr, length, prot: str):
+    '''Only x86_64.'''
+    saved_rax = pwndbg.regs.rax
```

Comment:
Can we either add `OnlyAmd64` decorator or a check for it in the code like `if pwndbg.arch != X86_64`?

Reply:
sure